UI/ProductionWnd.cpp

```diff
@@ -723,17 +734,27 @@ namespace {
                 popup->AddMenuItem(GG::MenuItem(UserString("PAUSE"),            false, false, pause_action));
             }
 
-            if (queue_row && queue_row->elem.allowed_imperial_stockpile_use) {
-                popup->AddMenuItem(GG::MenuItem(UserString("DISALLOW_IMPERIAL_PP_STOCKPILE_USE"), false, false, disallow_stockpile_action));
-            } else {
-                popup->AddMenuItem(GG::MenuItem(UserString("ALLOW_IMPERIAL_PP_STOCKPILE_USE"), false, false, allow_stockpile_action));
+            switch (build_type) {
+            case BT_BUILDING:
+            case BT_SHIP:
+                if (queue_row && queue_row->elem.allowed_imperial_stockpile_use) {
+                    popup->AddMenuItem(GG::MenuItem(UserString("DISALLOW_IMPERIAL_PP_STOCKPILE_USE"), false, false, disallow_stockpile_action));
+                } else {
+                    popup->AddMenuItem(GG::MenuItem(UserString("ALLOW_IMPERIAL_PP_STOCKPILE_USE"), false, false, allow_stockpile_action));
+                }
+                break;
+            case BT_STOCKPILE:
+                break;
             }
+
             // pedia lookup
             std::string item_name = "";
             if (build_type == BT_BUILDING) {
                 item_name = queue_row->elem.item.name;
             } else if (build_type == BT_SHIP) {
                 item_name = GetShipDesign(queue_row->elem.item.design_id)->Name(false);
+            } else if (build_type == BT_STOCKPILE) {
+                item_name = "PROJECT_BT_STOCKPILE";
```

Comment:
not sure if this is user visible, but in UI code, use a `UserString` call for stuff like this.